gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionFactory.java

```diff
@@ -572,22 +578,41 @@ public Schema schema() {
     }
   }
 
+  @Log
   public static class RecordConverter extends ComplexConverter {
-
+    private static final Logger LOG = LoggerFactory.getLogger(JsonIntermediateToAvroConverter.class);
     private HashMap<String, JsonElementConverter> converters = new HashMap<>();
     private JsonObject _schemaNode;
     private Schema _schema;
+    private long numFailedConversion = 0;
+    private State workUnit;
 
     public RecordConverter(String fieldName, boolean nullable, String sourceType, JsonObject schemaNode,
         WorkUnitState state, String namespace)
         throws UnsupportedDateTypeException {
       super(fieldName, nullable, sourceType);
+      workUnit = state;
       _schemaNode = schemaNode;
       _schema = buildRecordSchema(getValuesWithinDataType(_schemaNode).getAsJsonArray(), state,
           getPropOrBlankString(getDataType(_schemaNode), "name"), namespace);
     }
 
-    public Schema buildRecordSchema(JsonArray schema, WorkUnitState workUnit, String name, String namespace) {
+    public RecordConverter(String fieldName, boolean nullable, String sourceType, JsonArray schemaNodeArray,
```

Comment:
I'm a bit confused what this second constructor does. Is it necessary for creating a recordConverter from JsonIntermediateToString?

Reply:
Yes, the second constructor is redundant. Removed in future commits.